openedx/frontend-app-course-authoring
# [ROLES-26] Helper function for ingesting permission data

### Background 

This PR is to fetch the new (LoggedIn) User Permissions from the endpoint `/api/course_roles/v1/user_permissions/`.

### Changes done
- Add new function on generic data `api, slice, selector and thunk`
- Add a new hook for handling the permissions in the `CourseTeam` component.
- Add `UserPermissionsEnabledFlag` 
- Add specs


### Testing instructions
To test this code please use this branch for course-authoring and ROLES-2-course_roles_setup for edx-platform

1. Go to LMS admin and enable/create these waffle flags:
 - `contentstore.new_studio_mfe.use_new_course_team_page`
 - `course_roles.use_permission_checks`

2. Go to a Demo Course as a user with proper access to view CourseTeams in the CMS

3. **As LoggedIn Admin** (Same behavior as current):
  - You should see the *+ New team member* and *+ Add a new team member* buttons (Since we're using the `hasManageAllUsersPerm` as part of the condition [OR]).

4. **As a LoggedIn User**: 
- You'll have to disable `allow_actions` from this file `cms/djangoapps/contentstore/utils.py:1366` (you can hardcode this for testing).
- Enable `course_roles.use_permission_checks` flag enabled
- Enable the `manage_all_users` permission from this file `openedx/core/djangoapps/course_roles/views.py:49` (you can hardcode `'permissions': ['manage_all_users']`).
- You should still see:
  - *Course Team Members cards* 
  - *+ New team member* button
  - *+ Add a new team member* button
